### PR TITLE
Add missing fields to AuditEntry

### PR DESCRIPTION
**Pull request by patriknordlen**
Fixes: #2785.

Adds the following fields to the `AuditEntry` struct and populates them when they exist in a response from the API:

- `ActorIP` - IP of actor (if enabled in GitHub audit logs)
- `ActorLocation` - location of actor (country code etc)
- `HashedToken` - hashed token identifier when an action is performed using an access token
- `JobWorkflowRef` - reference to workflow job when a workflow runs
- `OAuthApplicationID` - OAuth application identifier when an OAuth application performs an action
- `OrgID` - organization numeric ID
- `ProgrammaticAccessType` - type of programmatic access (i.e. via token, cli, oauth app...)
- `PullRequestID` - self-explanatory
- `PullRequestTitle` - self-explanatory
- `PullRequestURL` - self-explanatory
- `Reasons` - Used when a user overrides branch protection, includes info on user's stated reason why
- `RunNumber` - the numerical ID of the workflow run
- `TokenID` - token ID when an action is performed using an access token
- `TokenScopes` - string containing scopes assigned to a token
- `Topic` - topic used in workflow runs
- `UserAgent` - user-agent used in performing an action

### CODE REVIEW

File: github/orgs_audit_log.go [hunk 1]
```diff
@@ -29,69 +29,96 @@ type HookConfig struct {
...
+	OrgID                  *int64                 `json:"org_id,omitempty"`
+	OverriddenCodes        []string               `json:"overridden_codes,omitempty"`
+	Permission             *string                `json:"permission,omitempty"` // The permission level for membership changes, for example `admin` or `read`.
+	PreviousVisibility     *string                `json:"previous_visibility,omitempty"`
+	ProgrammaticAccessType *string                `json:"programmatic_access_type,omitempty"`
+	PullRequestID          *int64                 `json:"pull_request_id,omitempty"`
+	PullRequestTitle       *string                `json:"pull_request_title,omitempty"`
+	PullRequestURL         *string                `json:"pull_request_url,omitempty"`
+	ReadOnly               *string                `json:"read_only,omitempty"`
+	Reasons                []PolicyOverrideReason `json:"reasons,omitempty"`
```
Comment (gmlewis) [inline github/orgs_audit_log.go:93]:
```suggestion
	Reasons                []*PolicyOverrideReason `json:"reasons,omitempty"`
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @patriknordlen!
Just a few tweaks, please, then we will be ready to merge.

File: github/orgs_audit_log.go [hunk 2]
```diff
@@ -29,69 +29,96 @@ type HookConfig struct {
 	Secret *string `json:"secret,omitempty"`
 }
 
+type ActorLocation struct {
```
[the remark this answers is not in the document]
Comment (patriknordlen) [reply github/orgs_audit_log.go:33]: Added!

**Comment by patriknordlen** [comment]
Resolving type-related issues in tests now.

**Review by gmlewis — APPROVED** [review]
Thank you, @patriknordlen !
LGTM.
Merging after tests pass.

**Pull request merged**